R/fb_player_scouting.R

```diff
@@ -53,20 +56,32 @@ fb_player_scouting_report <- function(player_url, pos_versus, time_pause=3) {
   scout_level1_url <- main_cats %>% rvest::html_nodes("ul li") %>%
     rvest::html_nodes("a") %>% rvest::html_attr("href") %>% paste0(main_url, .)
 
+  league_comp_names <- main_cats %>% rvest::html_nodes("ul li") %>% rvest::html_text() %>% stringr::str_squish()
+
+  if(!is.null(league_comp_name)) {
+    league_comp_name_regex <- league_comp_name %>% paste0(collapse = "|")
+
+    if(!any(grep(league_comp_name_regex, league_comp_names))) {
+      stop(glue::glue("'{league_comp_name}' not a valid value for `league_comp_name`. Please re-copy and paste the correct value(s)"))
+    } else {
+      scout_level1_url <- scout_level1_url[grep(league_comp_name_regex, league_comp_names)]
+    }
+  }
+
 
   all_scout_pos <- data.frame()
 
-  for(each_scout_url in scout_level1_url) {
+  for(each_scout_url in 1:length(scout_level1_url)) {
     Sys.sleep(time_pause)
 
-    scout_pg <- xml2::read_html(each_scout_url)
+    scout_pg <- .load_page(scout_level1_url[each_scout_url])
 
-    period <- scout_pg %>% rvest::html_nodes("#all_scout") %>% rvest::html_nodes(".section_heading_text") %>% rvest::html_text() %>%
-      unique() %>% stringr::str_squish()
+    period <- scout_pg %>% rvest::html_nodes(".filter") %>% .[1] %>% rvest::html_elements("div") %>% rvest::html_text() %>%
+      unique() %>% stringr::str_squish() %>% .[each_scout_url]
```

Comment:
This is essentially the fix for what was causing the initial issue, just a simple element name change

Reply:
👍